controllers/data.controller.js

```diff
@@ -256,14 +258,30 @@ exports.sendQuestionToFrontend = async (req, res) => {
         .status(httpStatus.NOT_FOUND)
         .json({ error: "No questions found for this tier" });
     }
+    let multipleChoices = [];
+    let tries = 0;
+    let selectedQuestion;
+    // tries to avoid infinite loops
+    while (multipleChoices.length < 4 && tries < 5) {
+      selectedQuestion =
+        questions[Math.floor(Math.random() * questions.length)];
 
-    const selectedQuestion =
-      questions[Math.floor(Math.random() * questions.length)];
+      multipleChoices = await getOtherMultipleChoiceAnswers(
+        selectedQuestion,
+        currentTier,
+      );
+      tries += 1;
+    }
+    if (multipleChoices.length < 4) {
```

Comment:
what is the function of this code..?

Reply:
I think maybe by mistake it was added

Reply:
> I think maybe by mistake it was added

The purpose is to throw an error of there is no suitable question to send as you ask in previous review